Handbrake qsv encode + HDR

### Problem Description

Hello, when I encode a 4K HDR film with handbrake, I sometimes only get artifacts via QuickSync or the picture looks very strange. If I do not encode via QuickSync (10bit) but with the x265 software encoder, then the picture is normal.
With Full HD films, I have not yet been able to determine the problem. 
I have seen this in 1.4.0, 1.4.1 and in nightly.


edit: i think i found the error.
if i decode with quicksync, then i have this error (but only with 4k HDR movies).
if i deactivated decode with quicksync, then i have not this error.

I have uploadet 2 screenshots, 
In first, you see the same movie with x265 Software encode
![x265soft](https://user-images.githubusercontent.com/10309056/129838153-403cd330-208b-499d-84d4-af5ed4ab73c3.jpg)
here is the same movie with x265 Hardware encode
![x265hard](https://user-images.githubusercontent.com/10309056/129838161-d23fdf54-4a27-4578-9f1c-d29cb61b906d.jpg)
and here without quicksync decode
![x265hardwqs](https://user-images.githubusercontent.com/10309056/129839746-c8e203d4-21ac-4c99-b978-20b322d262f6.jpg)





### Activity Log, Crash Log or any other details

```text
HandBrake 1.4.1 (2021081500)
OS: Microsoft Windows NT 10.0.19043.0
CPU: 11th Gen Intel(R) Core(TM) i5-11500 @ 2.70GHz
Ram: 15975 MB, 
GPU Information:
  Intel(R) UHD Graphics 750 - 30.0.100.9805
Screen: 1920x1080
Temp Dir: C:\Users\Thomas\AppData\Local\Temp\
Install Dir: C:\Program Files\HandBrake\
Data Dir: C:\Users\Thomas\AppData\Roaming\HandBrake

-------------------------------------------


# Starting Encode ...
[06:21:25] base preset: Intel HEVC (Modified)
Remote Process started with Process ID: 12012 using port: 8037. Max Allowed Instances: 1
Worker: Starting HandBrake Engine ...
Worker: Starting Web Server on port 8037 ...
Worker: Disconnected worker monitoring enabled!
[06:21:26] Compile-time hardening features are enabled
Cannot load nvEncodeAPI64.dll
Cannot load nvEncodeAPI64.dll
Cannot load nvEncodeAPI64.dll
Cannot load nvEncodeAPI64.dll
06:21:26] hb_init: starting libhb thread
[06:21:26] Starting work at: Wed Aug 18 06:21:26 2021
[06:21:26] 1 job(s) to process
[06:21:26] json job:
{
  "Audio": {
    "AudioList": [
      {
        "DRC": 0,
        "Encoder": "copy:ac3",
        "Gain": 0,
        "Mixdown": -1,
        "NormalizeMixLevel": false,
        "Samplerate": 0,
        "Name": "Surround 5.1",
        "Track": 0,
        "DitherMethod": 0
      }
    ],
    "CopyMask": [
      "copy:aac",
      "copy:ac3",
      "copy:dtshd",
      "copy:dts",
      "copy:eac3",
      "copy:flac",
      "copy:mp3",
      "copy:truehd",
      "copy:mp2"
    ],
    "FallbackEncoder": "ac3"
  },
  "Destination": {
    "ChapterList": [
      {
        "Name": "Chapter 1"
      },
      {
        "Name": "Chapter 2"
      },
      {
        "Name": "Chapter 3"
      },
      {
        "Name": "Chapter 4"
      },
      {
        "Name": "Chapter 5"
      },
      {
        "Name": "Chapter 6"
      },
      {
        "Name": "Chapter 7"
      },
      {
        "Name": "Chapter 8"
      },
      {
        "Name": "Chapter 9"
      },
      {
        "Name": "Chapter 10"
      },
      {
        "Name": "Chapter 11"
      },
      {
        "Name": "Chapter 12"
      },
      {
        "Name": "Chapter 13"
      },
      {
        "Name": "Chapter 14"
      },
      {
        "Name": "Chapter 15"
      },
      {
        "Name": "Chapter 16"
      }
    ],
    "ChapterMarkers": true,
    "AlignAVStart": false,
    "File": "E:\\BluRay\\Encode\\Mortal Kombat_t00-1.mkv",
    "Mp4Options": {
      "IpodAtom": false,
      "Mp4Optimize": false
    },
    "Mux": "av_mkv"
  },
  "Filters": {
    "FilterList": [
      {
        "ID": 4,
        "Settings": {
          "mode": "7"
        }
      },
      {
        "ID": 3,
        "Settings": {
          "block-height": "16",
          "block-thresh": "40",
          "block-width": "16",
          "filter-mode": "2",
          "mode": "3",
          "motion-thresh": "1",
          "spatial-metric": "2",
          "spatial-thresh": "1"
        }
      },
      {
        "ID": 13,
        "Settings": {
          "crop-bottom": "280",
          "crop-left": "0",
          "crop-right": "0",
          "crop-top": "280",
          "height": "1600",
          "width": "3840"
        }
      },
      {
        "ID": 6,
        "Settings": {
          "mode": "1"
        }
      }
    ]
  },
  "PAR": {
    "Num": 1,
    "Den": 1
  },
  "Metadata": {
    "Name": "Mortal Kombat"
  },
  "SequenceID": 0,
  "Source": {
    "Angle": 1,
    "Range": {
      "Type": "chapter",
      "Start": 1,
      "End": 1
    },
    "Title": 1,
    "Path": "E:\\BluRay\\RIP\\Mortal Kombat_t00.mkv"
  },
  "Subtitle": {
    "Search": {
      "Burn": false,
      "Default": false,
      "Enable": false,
      "Forced": false
    },
    "SubtitleList": [
      {
        "Burn": false,
        "Default": true,
        "Forced": false,
        "ID": 2,
        "Offset": 0,
        "Track": 1
      }
    ]
  },
  "Video": {
    "Encoder": "qsv_h265_10bit",
    "Level": "auto",
    "TwoPass": false,
    "Turbo": false,
    "ColorMatrixCode": 0,
    "Options": "ref=6:b-pyramid=1:tu=1:force-cqp=1:cqp-offset-i=2:cqp-offset-p=5:cqp-offset-b=10:lowpower=0",
    "Preset": "quality",
    "Profile": "main10",
    "Quality": 27,
    "QSV": {
      "Decode": true
    }
  }
}
[06:21:27] CPU: 11th Gen Intel(R) Core(TM) i5-11500 @ 2.70GHz
[06:21:27]  - logical processor count: 12
[06:21:27] Intel Quick Sync Video support: yes, gpu list: 0
[06:21:27] Intel Quick Sync Video integrated adapter with index 0
[06:21:27]  - Intel Media SDK hardware: API 1.35 (minimum: 1.3)
[06:21:27]  - Decode support:  h264 hevc (8bit: yes, 10bit: yes) av1 (8bit: yes, 10bit: yes)
[06:21:27]  - H.264 encoder: yes
[06:21:27]     - preferred implementation: hardware (1) via D3D11
[06:21:27]     - capabilities (hardware):  breftype icq+la+i+downs vsinfo opt1 opt2+mbbrc+extbrc+trellis+ib_adapt+nmpslice
[06:21:27]  - H.265 encoder: yes (8bit: yes, 10bit: yes)
[06:21:27]     - preferred implementation: hardware (1) via D3D11
[06:21:27]     - capabilities (hardware):  lowpower bpyramid icq vsinfo opt1
[06:21:27] hb_scan: path=E:\BluRay\RIP\Mortal Kombat_t00.mkv, title_index=1
udfread ERROR: ECMA 167 Volume Recognition failed
src/libbluray/disc/disc.c:333: failed opening UDF image E:\BluRay\RIP\Mortal Kombat_t00.mkv
src/libbluray/disc/disc.c:437: error opening file BDMV\index.bdmv
src/libbluray/disc/disc.c:437: error opening file BDMV\BACKUP\index.bdmv
src/libbluray/bluray.c:2646: nav_get_title_list(E:\BluRay\RIP\Mortal Kombat_t00.mkv\) failed
[06:21:27] bd: not a bd - trying as a stream/file instead
libdvdread: Encrypted DVD support unavailable.
libdvdread: DVDOpenFileUDF:UDFFindFile /VIDEO_TS/VIDEO_TS.IFO failed
libdvdread: DVDOpenFileUDF:UDFFindFile /VIDEO_TS/VIDEO_TS.BUP failed
[06:21:27] dvd: not a dvd - trying as a stream/file instead
Input #0, matroska,webm, from 'E:\BluRay\RIP\Mortal Kombat_t00.mkv':
  Metadata:
    title           : Mortal Kombat
    encoder         : libmakemkv v1.16.4 (1.3.10/1.5.2) win(x64-release)
    creation_time   : 2021-07-27T15:38:26.000000Z
  Duration: 01:50:03.64, start: 0.000000, bitrate: 60405 kb/s
  Chapters:
    Chapter #0:0: start 0.000000, end 154.863042
      Metadata:
        title           : Chapter 01
    Chapter #0:1: start 154.863042, end 445.403292
      Metadata:
        title           : Chapter 02
    Chapter #0:2: start 445.403292, end 773.773000
      Metadata:
        title           : Chapter 03
    Chapter #0:3: start 773.773000, end 1153.569083
      Metadata:
        title           : Chapter 04
    Chapter #0:4: start 1153.569083, end 1644.726417
      Metadata:
        title           : Chapter 05
    Chapter #0:5: start 1644.726417, end 2087.668917
      Metadata:
        title           : Chapter 06
    Chapter #0:6: start 2087.668917, end 2533.697833
      Metadata:
        title           : Chapter 07
    Chapter #0:7: start 2533.697833, end 2859.565042
      Metadata:
        title           : Chapter 08
    Chapter #0:8: start 2859.565042, end 3399.270875
      Metadata:
        title           : Chapter 09
    Chapter #0:9: start 3399.270875, end 3949.278667
      Metadata:
        title           : Chapter 10
    Chapter #0:10: start 3949.278667, end 4384.129750
      Metadata:
        title           : Chapter 11
    Chapter #0:11: start 4384.129750, end 4763.091667
      Metadata:
        title           : Chapter 12
    Chapter #0:12: start 4763.091667, end 5237.690792
      Metadata:
        title           : Chapter 13
    Chapter #0:13: start 5237.690792, end 5745.030958
      Metadata:
        title           : Chapter 14
    Chapter #0:14: start 5745.030958, end 5938.599333
      Metadata:
        title           : Chapter 15
    Chapter #0:15: start 5938.599333, end 6603.638708
      Metadata:
        title           : Chapter 16
  Stream #0:0(eng): Video: hevc (Main 10), yuv420p10le(tv, bt2020nc/bt2020/smpte2084), 3840x2160 [SAR 1:1 DAR 16:9], 23.98 fps, 23.98 tbr, 1k tbn, 23.98 tbc
    Metadata:
      BPS-eng         : 59736205
      DURATION-eng    : 01:50:03.638708333
      NUMBER_OF_FRAMES-eng: 158329
      NUMBER_OF_BYTES-eng: 49309534622
      SOURCE_ID-eng   : 001011
      _STATISTICS_WRITING_APP-eng: MakeMKV v1.16.4 win(x64-release)
      _STATISTICS_WRITING_DATE_UTC-eng: 2021-07-27 15:38:26
      _STATISTICS_TAGS-eng: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES SOURCE_ID
  Stream #0:1(deu): Audio: ac3, 48000 Hz, 5.1(side), fltp, 640 kb/s (default)
    Metadata:
      title           : Surround 5.1
      BPS-eng         : 640000
      DURATION-eng    : 01:50:03.648000000
      NUMBER_OF_FRAMES-eng: 206364
      NUMBER_OF_BYTES-eng: 528291840
      SOURCE_ID-eng   : 001102
      _STATISTICS_WRITING_APP-eng: MakeMKV v1.16.4 win(x64-release)
      _STATISTICS_WRITING_DATE_UTC-eng: 2021-07-27 15:38:26
      _STATISTICS_TAGS-eng: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES SOURCE_ID
  Stream #0:2(deu): Subtitle: hdmv_pgs_subtitle
    Metadata:
      BPS-eng         : 28412
      DURATION-eng    : 01:40:45.476937500
      NUMBER_OF_FRAMES-eng: 2008
      NUMBER_OF_BYTES-eng: 21471022
      SOURCE_ID-eng   : 0012A2
      _STATISTICS_WRITING_APP-eng: MakeMKV v1.16.4 win(x64-release)
      _STATISTICS_WRITING_DATE_UTC-eng: 2021-07-27 15:38:26
      _STATISTICS_TAGS-eng: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES SOURCE_ID
  Stream #0:3(deu): Subtitle: hdmv_pgs_subtitle
    Metadata:
      BPS-eng         : 1326
      DURATION-eng    : 01:34:21.593437500
      NUMBER_OF_FRAMES-eng: 86
      NUMBER_OF_BYTES-eng: 938662
      SOURCE_ID-eng   : 0012AE
      _STATISTICS_WRITING_APP-eng: MakeMKV v1.16.4 win(x64-release)
      _STATISTICS_WRITING_DATE_UTC-eng: 2021-07-27 15:38:26
      _STATISTICS_TAGS-eng: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES SOURCE_ID
[06:21:27] scan: decoding previews for title 1
[06:21:27] scan: audio 0x1: ac3, rate=48000Hz, bitrate=640000 Deutsch (AC3) (5.1 ch) (640 kbps)
[06:21:30] scan: 10 previews, 3840x2160, 23.976 fps, autocrop = 276/276/0/0, aspect 16:9, PAR 1:1, color profile: 9-16-9
[06:21:30] scan: mastering display metadata: r(0.6800,0.3200) g(0.2650,0.6900) b(0.1500 0.0600) wp(0.3127, 0.3290) min_luminance=0.000100, max_luminance=1000.000000
[06:21:30] scan: content light level: max_cll=825, max_fall=118
[06:21:30] scan: supported video decoders: avcodec qsv
[06:21:30] libhb: scan thread found 1 valid title(s)
[06:21:30] Starting Task: Encoding Pass
[06:21:30] work: only 1 chapter, disabling chapter markers
[06:21:30] job configuration:
[06:21:30]  * source
[06:21:30]    + E:\BluRay\RIP\Mortal Kombat_t00.mkv
[06:21:30]    + title 1, chapter(s) 1 to 1
[06:21:30]    + container: matroska,webm
[06:21:30]    + data rate: 60405 kbps
[06:21:30]  * destination
[06:21:30]    + E:\BluRay\Encode\Mortal Kombat_t00-1.mkv
[06:21:30]    + container: Matroska (libavformat)
[06:21:30]  * video track
[06:21:30]    + decoder: hevc_qsv 8-bit
[06:21:30]    + filters
[06:21:30]      + Comb Detect (mode=3:spatial-metric=2:motion-thresh=1:spatial-thresh=1:filter-mode=2:block-thresh=40:block-width=16:block-height=16)
[06:21:30]      + Decomb (mode=39)
[06:21:30]      + Framerate Shaper (mode=1)
[06:21:30]        + frame rate: 23.976 fps -> constant 23.976 fps
[06:21:30]      + Crop and Scale (width=3840:height=1600:crop-top=280:crop-bottom=280:crop-left=0:crop-right=0)
[06:21:30]        + source: 3840 * 2160, crop (280/280/0/0): 3840 * 1600, scale: 3840 * 1600
[06:21:30]    + Output geometry
[06:21:30]      + storage dimensions: 3840 x 1600
[06:21:30]      + pixel aspect ratio: 1 : 1
[06:21:30]      + display dimensions: 3840 x 1600
[06:21:30]    + encoder: H.265 10-bit (Intel Media SDK)
[06:21:30]      + preset:  quality
[06:21:30]      + options: ref=6:b-pyramid=1:tu=1:force-cqp=1:cqp-offset-i=2:cqp-offset-p=5:cqp-offset-b=10:lowpower=0
[06:21:30]      + profile: main10
[06:21:30]      + level:   auto
[06:21:30]      + quality: 27.00 (ICQ)
[06:21:30]      + color profile: 9-16-9
[06:21:30]      + mastering display metadata: r(0.6800,0.3200) g(0.2650,0.6900) b(0.1500 0.0600) wp(0.3127, 0.3290) min_luminance=0.000100, max_luminance=1000.000000
[06:21:30]      + content light level: max_cll=825, max_fall=118
[06:21:30]  * subtitle track 1, Deutsch [PGS] (track 1, id 0x3, Picture) -> Passthrough, Default
[06:21:30]  * audio track 1
[06:21:30]    + name: Surround 5.1
[06:21:30]    + decoder: Deutsch (AC3) (5.1 ch) (640 kbps) (track 1, id 0x1)
[06:21:30]      + bitrate: 640 kbps, samplerate: 48000 Hz
[06:21:30]    + AC3 Passthru
[06:21:30] sync: expecting 3712 video frames
[06:21:30] qsv: integrated qsv adapter with index 0 has been selected
[06:21:30] encqsvInit: using encode-only path
[06:21:30] encqsvInit: H.265/HEVC Main 10 profile @ level (null)
[06:21:30] encqsvInit: TargetUsage 1 AsyncDepth 6
[06:21:30] encqsvInit: GopRefDist 4 GopPicSize 32 NumRefFrame 6
[06:21:30] encqsvInit: BFramesMax 3 BRefType pyramid
[06:21:30] encqsvInit: RateControlMethod CQP with QPI 29 QPP 32 QPB 37
[06:21:32] sync: first pts video is 0
[06:21:32] sync: "Chapter 1" (1) at frame 1 time 0
[06:21:32] sync: first pts audio 0x1 is 0
[06:21:32] qsv_enc_init: using 'hardware (1) via D3D11' implementation, API: 1.35
[06:23:26] sync: first pts subtitle 0x3 is 5600610
[06:26:42] reader: end of chapter 1 (media 1) reached at media chapter 2
[06:26:42] reader: done. 1 scr changes
[06:26:44] sync: "Chapter 2" (2) at frame 3715 time 13956483
[06:26:48] work: average encoding speed for job is 11.732439 fps
[06:26:48] comb detect: heavy 2779 | light 456 | uncombed 480 | total 3715
[06:26:48] decomb: deinterlaced 2779 | blended 456 | unfiltered 480 | total 3715
[06:26:48] vfr: 3719 frames output, 0 dropped and 4 duped for CFR/PFR
[06:26:48] vfr: lost time: 0 (0 frames)
[06:26:48] vfr: gained time: 0 (0 frames) (0 not accounted for)
[06:26:48] ac3-decoder done: 4842 frames, 0 decoder errors
[06:26:48] hevc_qsv-decoder done: 3715 frames, 0 decoder errors
[06:26:48] sync: got 3715 frames, 3712 expected
[06:26:48] sync: framerate min 4.785 fps, max 23.976 fps, avg 23.950 fps
[06:26:48] mux: track 0, 3719 frames, 218510018 bytes, 11269.70 kbps, fifo 512
[06:26:48] mux: track 1, 4842 frames, 12395520 bytes, 639.30 kbps, fifo 512
[06:26:48] mux: track 2, 12 frames, 138216 bytes, 7.13 kbps, fifo 16
[06:26:48] Finished work at: Wed Aug 18 06:26:48 2021
[06:26:48] libhb: work result = 0

# Job Completed!
```


### What Operating System are you running?

Windows 10

### What version of HandBrake are you running?

1.4.1

### Where did you download HandBrake from?

_No response_

Is it possible that every movie decode with 8 bit instead of 8 bit or 10 bit?
I have 10 bit movie and it will be decoded with 8 bit. Thats strange.

```
HandBrake 1.4.1 (2021081500)
OS: Microsoft Windows NT 10.0.19043.0
CPU: 11th Gen Intel(R) Core(TM) i5-11500 @ 2.70GHz
Ram: 15975 MB, 
GPU Information:
  Intel(R) UHD Graphics 750 - 30.0.100.9805
Screen: 1920x1080
Temp Dir: C:\Users\Thomas\AppData\Local\Temp\
Install Dir: C:\Program Files\HandBrake\
Data Dir: C:\Users\Thomas\AppData\Roaming\HandBrake

-------------------------------------------


# Starting Encode ...
[07:39:51] base preset: Intel HEVC (Modified)
Remote Process started with Process ID: 5916 using port: 8037. Max Allowed Instances: 1
Worker: Starting HandBrake Engine ...
Worker: Starting Web Server on port 8037 ...
Worker: Disconnected worker monitoring enabled!
[07:39:51] Compile-time hardening features are enabled
Cannot load nvEncodeAPI64.dll
Cannot load nvEncodeAPI64.dll
Cannot load nvEncodeAPI64.dll
Cannot load nvEncodeAPI64.dll
07:39:52] hb_init: starting libhb thread
[07:39:52] Starting work at: Wed Aug 25 07:39:52 2021
[07:39:52] 1 job(s) to process
[07:39:52] json job:
{
  "Audio": {
    "AudioList": [
      {
        "DRC": 0,
        "Encoder": "fdk_aac",
        "Gain": 0,
        "Mixdown": 7,
        "NormalizeMixLevel": false,
        "Quality": 4,
        "Samplerate": 0,
        "Name": "Surround 5.1",
        "Track": 0,
        "DitherMethod": 0
      }
    ],
    "CopyMask": [
      "copy:aac",
      "copy:ac3",
      "copy:dtshd",
      "copy:dts",
      "copy:eac3",
      "copy:flac",
      "copy:mp3",
      "copy:truehd",
      "copy:mp2"
    ],
    "FallbackEncoder": "ac3"
  },
  "Destination": {
    "ChapterList": [
      {
        "Name": "Chapter 1"
      },
      {
        "Name": "Chapter 2"
      },
      {
        "Name": "Chapter 3"
      },
      {
        "Name": "Chapter 4"
      },
      {
        "Name": "Chapter 5"
      },
      {
        "Name": "Chapter 6"
      },
      {
        "Name": "Chapter 7"
      },
      {
        "Name": "Chapter 8"
      },
      {
        "Name": "Chapter 9"
      },
      {
        "Name": "Chapter 10"
      },
      {
        "Name": "Chapter 11"
      },
      {
        "Name": "Chapter 12"
      }
    ],
    "ChapterMarkers": true,
    "AlignAVStart": false,
    "File": "E:\\BluRay\\Encode\\34John Wick_t00.mkv",
    "Mp4Options": {
      "IpodAtom": false,
      "Mp4Optimize": false
    },
    "Mux": "av_mkv"
  },
  "Filters": {
    "FilterList": [
      {
        "ID": 4,
        "Settings": {
          "mode": "7"
        }
      },
      {
        "ID": 3,
        "Settings": {
          "block-height": "16",
          "block-thresh": "40",
          "block-width": "16",
          "filter-mode": "2",
          "mode": "3",
          "motion-thresh": "1",
          "spatial-metric": "2",
          "spatial-thresh": "1"
        }
      },
      {
        "ID": 13,
        "Settings": {
          "crop-bottom": "280",
          "crop-left": "0",
          "crop-right": "0",
          "crop-top": "280",
          "height": "1600",
          "width": "3840"
        }
      },
      {
        "ID": 6,
        "Settings": {
          "mode": "1"
        }
      }
    ]
  },
  "PAR": {
    "Num": 1,
    "Den": 1
  },
  "Metadata": {
    "Name": "John Wick"
  },
  "SequenceID": 0,
  "Source": {
    "Angle": 1,
    "Range": {
      "Type": "chapter",
      "Start": 1,
      "End": 1
    },
    "Title": 1,
    "Path": "E:\\BluRay\\RIP\\John Wick_t00.mkv"
  },
  "Subtitle": {
    "Search": {
      "Burn": false,
      "Default": false,
      "Enable": false,
      "Forced": false
    },
    "SubtitleList": [
      {
        "Burn": false,
        "Default": true,
        "Forced": false,
        "ID": 3,
        "Offset": 0,
        "Track": 2
      }
    ]
  },
  "Video": {
    "Encoder": "qsv_h265_10bit",
    "Level": "auto",
    "TwoPass": false,
    "Turbo": false,
    "ColorMatrixCode": 0,
    "Options": "ref=6:b-pyramid=1:tu=1:force-cqp=1:cqp-offset-i=2:cqp-offset-p=5:cqp-offset-b=10:lowpower=0",
    "Preset": "quality",
    "Profile": "main10",
    "Quality": 34,
    "QSV": {
      "Decode": false
    }
  }
}
[07:39:52] CPU: 11th Gen Intel(R) Core(TM) i5-11500 @ 2.70GHz
[07:39:52]  - logical processor count: 12
[07:39:52] Intel Quick Sync Video support: yes, gpu list: 0
[07:39:52] Intel Quick Sync Video integrated adapter with index 0
[07:39:52]  - Intel Media SDK hardware: API 1.35 (minimum: 1.3)
[07:39:52]  - Decode support:  h264 hevc (8bit: yes, 10bit: yes) av1 (8bit: yes, 10bit: yes)
[07:39:52]  - H.264 encoder: yes
[07:39:52]     - preferred implementation: hardware (1) via D3D11
[07:39:52]     - capabilities (hardware):  breftype icq+la+i+downs vsinfo opt1 opt2+mbbrc+extbrc+trellis+ib_adapt+nmpslice
[07:39:52]  - H.265 encoder: yes (8bit: yes, 10bit: yes)
[07:39:52]     - preferred implementation: hardware (1) via D3D11
[07:39:52]     - capabilities (hardware):  lowpower bpyramid icq vsinfo opt1
[07:39:52] hb_scan: path=E:\BluRay\RIP\John Wick_t00.mkv, title_index=1
udfread ERROR: ECMA 167 Volume Recognition failed
src/libbluray/disc/disc.c:333: failed opening UDF image E:\BluRay\RIP\John Wick_t00.mkv
src/libbluray/disc/disc.c:437: error opening file BDMV\index.bdmv
src/libbluray/disc/disc.c:437: error opening file BDMV\BACKUP\index.bdmv
src/libbluray/bluray.c:2646: nav_get_title_list(E:\BluRay\RIP\John Wick_t00.mkv\) failed
[07:39:52] bd: not a bd - trying as a stream/file instead
libdvdread: Encrypted DVD support unavailable.
libdvdread: DVDOpenFileUDF:UDFFindFile /VIDEO_TS/VIDEO_TS.IFO failed
libdvdread: DVDOpenFileUDF:UDFFindFile /VIDEO_TS/VIDEO_TS.BUP failed
[07:39:52] dvd: not a dvd - trying as a stream/file instead
Input #0, matroska,webm, from 'E:\BluRay\RIP\John Wick_t00.mkv':
  Metadata:
    title           : John Wick
    encoder         : libmakemkv v1.16.4 (1.3.10/1.5.2) win(x64-release)
    creation_time   : 2021-08-11T18:25:07.000000Z
  Duration: 01:41:31.68, start: 0.000000, bitrate: 63201 kb/s
  Chapters:
    Chapter #0:0: start 0.000000, end 395.437104
      Metadata:
        title           : Chapter 01
    Chapter #0:1: start 395.437104, end 850.684017
      Metadata:
        title           : Chapter 02
    Chapter #0:2: start 850.684017, end 1303.428428
      Metadata:
        title           : Chapter 03
    Chapter #0:3: start 1303.428428, end 2135.260260
      Metadata:
        title           : Chapter 04
    Chapter #0:4: start 2135.260260, end 2673.506840
      Metadata:
        title           : Chapter 05
    Chapter #0:5: start 2673.506840, end 3321.738405
      Metadata:
        title           : Chapter 06
    Chapter #0:6: start 3321.738405, end 3693.651985
      Metadata:
        title           : Chapter 07
    Chapter #0:7: start 3693.651985, end 4475.141808
      Metadata:
        title           : Chapter 08
    Chapter #0:8: start 4475.141808, end 4771.980314
      Metadata:
        title           : Chapter 09
    Chapter #0:9: start 4771.980314, end 5108.733734
      Metadata:
        title           : Chapter 10
    Chapter #0:10: start 5108.733734, end 5681.014348
      Metadata:
        title           : Chapter 11
    Chapter #0:11: start 5681.014348, end 6091.675008
      Metadata:
        title           : Chapter 12
  Stream #0:0(eng): Video: hevc (Main 10), yuv420p10le(tv, bt2020nc/bt2020/smpte2084), 3840x2160 [SAR 1:1 DAR 16:9], 23.98 fps, 23.98 tbr, 1k tbn, 23.98 tbc
    Metadata:
      BPS-eng         : 61614176
      DURATION-eng    : 01:41:31.675008340
      NUMBER_OF_FRAMES-eng: 146054
      NUMBER_OF_BYTES-eng: 46916692361
      SOURCE_ID-eng   : 001011
      _STATISTICS_WRITING_APP-eng: MakeMKV v1.16.4 win(x64-release)
      _STATISTICS_WRITING_DATE_UTC-eng: 2021-08-11 18:25:07
      _STATISTICS_TAGS-eng: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES SOURCE_ID
  Stream #0:1(deu): Audio: dts (DTS), 48000 Hz, 5.1(side), fltp, 1536 kb/s (default)
    Metadata:
      title           : Surround 5.1
      BPS-eng         : 1509000
      DURATION-eng    : 01:41:31.658666666
      NUMBER_OF_FRAMES-eng: 571093
      NUMBER_OF_BYTES-eng: 1149039116
      SOURCE_ID-eng   : 001100
      _STATISTICS_WRITING_APP-eng: MakeMKV v1.16.4 win(x64-release)
      _STATISTICS_WRITING_DATE_UTC-eng: 2021-08-11 18:25:07
      _STATISTICS_TAGS-eng: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES SOURCE_ID
  Stream #0:2(deu): Subtitle: hdmv_pgs_subtitle
    Metadata:
      BPS-eng         : 15360
      DURATION-eng    : 01:33:43.227394060
      NUMBER_OF_FRAMES-eng: 1101
      NUMBER_OF_BYTES-eng: 10796751
      SOURCE_ID-eng   : 0012A0
      _STATISTICS_WRITING_APP-eng: MakeMKV v1.16.4 win(x64-release)
      _STATISTICS_WRITING_DATE_UTC-eng: 2021-08-11 18:25:07
      _STATISTICS_TAGS-eng: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES SOURCE_ID
  Stream #0:3(deu): Subtitle: hdmv_pgs_subtitle
    Metadata:
      BPS-eng         : 62938
      DURATION-eng    : 01:37:04.929095762
      NUMBER_OF_FRAMES-eng: 3220
      NUMBER_OF_BYTES-eng: 45826877
      SOURCE_ID-eng   : 0012A1
      _STATISTICS_WRITING_APP-eng: MakeMKV v1.16.4 win(x64-release)
      _STATISTICS_WRITING_DATE_UTC-eng: 2021-08-11 18:25:07
      _STATISTICS_TAGS-eng: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES SOURCE_ID
  Stream #0:4(deu): Subtitle: hdmv_pgs_subtitle
    Metadata:
      BPS-eng         : 1639
      DURATION-eng    : 01:14:35.871705037
      NUMBER_OF_FRAMES-eng: 131
      NUMBER_OF_BYTES-eng: 917294
      SOURCE_ID-eng   : 0012A2
      _STATISTICS_WRITING_APP-eng: MakeMKV v1.16.4 win(x64-release)
      _STATISTICS_WRITING_DATE_UTC-eng: 2021-08-11 18:25:07
      _STATISTICS_TAGS-eng: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES SOURCE_ID
[07:39:52] scan: decoding previews for title 1
[07:39:52] scan: audio 0x1: dca, rate=48000Hz, bitrate=1536000 Deutsch (DTS) (5.1 ch) (1536 kbps)
[07:39:55] scan: 10 previews, 3840x2160, 23.976 fps, autocrop = 280/280/2/2, aspect 16:9, PAR 1:1, color profile: 9-16-9
[07:39:55] scan: mastering display metadata: r(0.6800,0.3200) g(0.2640,0.6900) b(0.1500 0.0600) wp(0.3127, 0.3290) min_luminance=0.000100, max_luminance=1000.000000
[07:39:55] scan: content light level: max_cll=1000, max_fall=400
[07:39:55] scan: supported video decoders: avcodec qsv
[07:39:55] libhb: scan thread found 1 valid title(s)
[07:39:55] Starting Task: Encoding Pass
[07:39:55] work: track 1, dithering not supported by codec
[07:39:55] work: only 1 chapter, disabling chapter markers
[07:39:55] job configuration:
[07:39:55]  * source
[07:39:55]    + E:\BluRay\RIP\John Wick_t00.mkv
[07:39:55]    + title 1, chapter(s) 1 to 1
[07:39:55]    + container: matroska,webm
[07:39:55]    + data rate: 63201 kbps
[07:39:55]  * destination
[07:39:55]    + E:\BluRay\Encode\34John Wick_t00.mkv
[07:39:55]    + container: Matroska (libavformat)
[07:39:55]  * video track
[07:39:55]    + decoder: hevc 8-bit
[07:39:55]    + filters
[07:39:55]      + Comb Detect (mode=3:spatial-metric=2:motion-thresh=1:spatial-thresh=1:filter-mode=2:block-thresh=40:block-width=16:block-height=16)
[07:39:55]      + Decomb (mode=39)
[07:39:55]      + Framerate Shaper (mode=1)
[07:39:55]        + frame rate: 23.976 fps -> constant 23.976 fps
[07:39:55]      + Crop and Scale (width=3840:height=1600:crop-top=280:crop-bottom=280:crop-left=0:crop-right=0)
[07:39:55]        + source: 3840 * 2160, crop (280/280/0/0): 3840 * 1600, scale: 3840 * 1600
[07:39:55]    + Output geometry
[07:39:55]      + storage dimensions: 3840 x 1600
[07:39:55]      + pixel aspect ratio: 1 : 1
[07:39:55]      + display dimensions: 3840 x 1600
[07:39:55]    + encoder: H.265 10-bit (Intel Media SDK)
[07:39:55]      + preset:  quality
[07:39:55]      + options: ref=6:b-pyramid=1:tu=1:force-cqp=1:cqp-offset-i=2:cqp-offset-p=5:cqp-offset-b=10:lowpower=0
[07:39:55]      + profile: main10
[07:39:55]      + level:   auto
[07:39:55]      + quality: 34.00 (ICQ)
[07:39:55]      + color profile: 9-16-9
[07:39:55]      + mastering display metadata: r(0.6800,0.3200) g(0.2640,0.6900) b(0.1500 0.0600) wp(0.3127, 0.3290) min_luminance=0.000100, max_luminance=1000.000000
[07:39:55]      + content light level: max_cll=1000, max_fall=400
[07:39:55]  * subtitle track 1, Deutsch [PGS] (track 2, id 0x4, Picture) -> Passthrough, Default
[07:39:55]  * audio track 1
[07:39:55]    + name: Surround 5.1
[07:39:55]    + decoder: Deutsch (DTS) (5.1 ch) (1536 kbps) (track 1, id 0x1)
[07:39:55]      + bitrate: 1536 kbps, samplerate: 48000 Hz
[07:39:55]    + mixdown: 5.1 Channels
[07:39:55]    + encoder: AAC (libfdk_aac)
[07:39:55]      + quality: 4.00, samplerate: 48000 Hz
[07:39:55] sync: expecting 9481 video frames
[07:39:55] qsv: integrated qsv adapter with index 0 has been selected
[07:39:55] encqsvInit: using encode-only path
[07:39:55] encqsvInit: H.265/HEVC Main 10 profile @ level (null)
[07:39:55] encqsvInit: TargetUsage 1 AsyncDepth 6
[07:39:55] encqsvInit: GopRefDist 4 GopPicSize 32 NumRefFrame 6
[07:39:55] encqsvInit: BFramesMax 3 BRefType pyramid
[07:39:55] encqsvInit: RateControlMethod CQP with QPI 36 QPP 39 QPB 44
[07:39:56] sync: first pts video is 0
[07:39:56] sync: "Chapter 1" (1) at frame 1 time 0
[07:39:56] sync: first pts audio 0x1 is 3690
[07:39:56] qsv_enc_init: using 'hardware (1) via D3D11' implementation, API: 1.35

```

--- a/contrib/ffmpeg/A27-qsv-fix-decode-10bit-hdr.patch
+++ b/contrib/ffmpeg/A27-qsv-fix-decode-10bit-hdr.patch
@@ -1,0 +1,28 @@
+diff --git a/libavcodec/hevc_mp4toannexb_bsf.c b/libavcodec/hevc_mp4toannexb_bsf.c
+index 790dfb0394..3b3732bbd0 100644
+--- a/libavcodec/hevc_mp4toannexb_bsf.c
++++ b/libavcodec/hevc_mp4toannexb_bsf.c
+@@ -121,7 +121,7 @@ static int hevc_mp4toannexb_filter(AVBSFContext *ctx, AVPacket *out)
+     HEVCBSFContext *s = ctx->priv_data;
+     AVPacket *in;
+     GetByteContext gb;
+-
++    int has_sps = 0, has_pps = 0;
+     int got_irap = 0;
+     int i, ret = 0;
+ 
+@@ -155,10 +155,13 @@ static int hevc_mp4toannexb_filter(AVBSFContext *ctx, AVPacket *out)
+         }
+ 
+         nalu_type = (bytestream2_peek_byte(&gb) >> 1) & 0x3f;
++        has_sps = (has_sps || nalu_type == HEVC_NAL_SPS);
++        has_pps = (has_pps || nalu_type == HEVC_NAL_PPS);
+ 
+         /* prepend extradata to IRAP frames */
+         is_irap       = nalu_type >= 16 && nalu_type <= 23;
+-        add_extradata = is_irap && !got_irap;
++        /* ignore the extradata if IRAP frame has sps and pps */
++        add_extradata = is_irap && !got_irap && !(has_sps && has_pps);
+         extra_size    = add_extradata * ctx->par_out->extradata_size;
+         got_irap     |= is_irap;
+ 

